scripts/test.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2019 Advanced Micro Devices, Inc. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at:
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+GFX reconstruct build script
+'''
+
+import argparse
+import copy
+import distutils.version
+import importlib
+import os
+import platform
+import re
+import shutil
+import subprocess
+import sys
+
+
+def is_windows():
+    '''
+    Check if the system is Windows
+    '''
+    return 'windows' == platform.system().lower()
+
+
+BUILD_ROOT = os.path.abspath(os.path.join(
+    os.path.split(os.path.abspath(__file__))[0], '..'))
+ALL_TESTS = {
+    'gfxrecon_application_test': [],
+    'gfxrecon_decode_test': [],
+    'gfxrecon_encode_test': [],
+    'gfxrecon_format_test': [],
+    'gfxrecon_util_test': [],
+    'VkLayer_gfxreconstruct_test': [],
+}
+
+
+class TestError(Exception):
+    '''
+    Raised on a test error
+    '''
+
+
+def import_build_script():
```

Comment:
I'm not a fan of unusual `import` workarounds; but there are situations where such are needed.  I don't think this is one of them...

Can you help me explain why this approach was taken over something more familiar like this in the `import` section above:
```
sys.path.insert(...)
import build
```
which would have allowed `build.XXX` to be used throughout the script, instead of having to pass `build_script` around as a parameter, simplifying the logic and making it more familiar... if there are good reasons, I'm all for it; but if there aren't, I prefer the more familiar construction.

Reply:
The sys.path.insert changes import paths for any script called from this one, or any other python script called in a subprocess.
I'm not a fan of changing the environment for the whole process tree if not needed.
This approach allows specific imports in the repo file tree + it is cleaner, since all import statements are grouped together at the top of the file.

Reply:
I don't entirely agree (IMO there is only negligible potential for impacts to subprocesses or other imported scripts, while confusion over the unusual import has already impacted at least one reviewer), but your motives are defensible.  In general, I'd like to see doc strings and/or comments explaining unusual constructions like this, so the reader doesn't have to guess at why they're being done this way.